runtime/src/main/java/org/corfudb/runtime/view/stream/AbstractContextStreamView.java

```diff
@@ -284,7 +283,7 @@ protected boolean doesEntryUpdateContext(final ILogData data) {
      */
     private void updatePointer(final ILogData data) {
         // Update the global pointer, if it is non-checkpoint data.
-        if (data.getType() == DataType.DATA && !data.hasCheckpointMetadata()) {
+        if (!data.hasCheckpointMetadata()) {
```

Comment:
Any reason data.getType() == DataType.DATA is removed?

Reply:
No, an earlier implementation needed to account for holes but actually this can be back in as it is adjusted to work as before. Nice catch.